### DISCUSSION TITLE
JS SDK readByQuery with relation

### DISCUSSION THREAD

**Discussion by NilsBaumgartner1994** [directus/directus#13144] [Archive]
``Same question as: https://github.com/directus/directus/discussions/5499

I want to request mealoffers which have a relation to meal.
```
await directus.items("mealoffers").readByQuery({
			fields: ['*', 'meals.*']
		});
```

I wish an output like this:

```
{
   title: 'test',
   meal: {
      id: 3,
      title: 'Test'
   }
}
```

**Comment by azrikahar** [comment]
If the field is `meal` in `mealsoffers` collection, then it should be `meal.*` (no `s`) instead of `meals.*` 👍

Additional note: Make sure the necessary read permission is added to the related collection(s) when doing this as well.

**Reply by NilsBaumgartner1994** [reply]
I achieved it by using the graphQL from directus:
It would be nice to see the docs updated in the SDK section on how to use relations with GraphQL

```
answer = await directus.graphql.items("query {\n" +
	"   mealoffers {\n" +
	"      id\n" +
	"      date\n" +
	"      meal {\n" +
	"      \tid\n" +
	"      \tname\n" +
	"      }\n" +
	"   }\n" +
	"}")
```
